packages/web3-eth-contract/src/index.js

```diff
@@ -784,6 +783,24 @@ Contract.prototype._on = function(){
     return subscription;
 };
 
+const filterAllEventsResults = (subOptions, data) => {
```

Comment:
You mentioned previously that this function didn't work 100% of the time, has that changed?

Reply:
This function works only for allEvents. Because awe can’t filter events by topics when ‘allEvents’